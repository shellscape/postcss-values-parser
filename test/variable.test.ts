/*
  Copyright Â© 2025 Andrew Powell

  This Source Code Form is subject to the terms of the Mozilla Public
  License, v. 2.0. If a copy of the MPL was not distributed with this
  file, You can obtain one at http://mozilla.org/MPL/2.0/.

  The above copyright notice and this permission notice shall be
  included in all copies or substantial portions of this Source Code Form.
*/
import { describe, expect, it } from 'vitest';

import { nodeToString, parse } from '../src/index.js';
import { snapshot, throws } from './fixtures/variable.json';

describe('variable parsing', () => {
  for (const fixture of snapshot) {
    it(`should parse: ${fixture}`, () => {
      const root = parse(fixture);
      const nodes = root.nodes.map((node) => {
        delete node.parent; // eslint-disable-line no-param-reassign
        return node;
      });
      const string = nodeToString(root);

      expect(root.first?.toString()).toMatchSnapshot();
      expect(string).toMatchSnapshot();
      expect(nodes).toMatchSnapshot();
    });
  }

  if (throws) {
    for (const fixture of throws) {
      it(`should throw for: ${fixture}`, () => {
        expect(() => parse(fixture)).toThrow();
      });
    }
  }
});
