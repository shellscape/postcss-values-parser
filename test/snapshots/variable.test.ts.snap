// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`variable parsing > should parse: --batman 1`] = `"--batman"`;

exports[`variable parsing > should parse: --batman 2`] = `"--batman"`;

exports[`variable parsing > should parse: --batman 3`] = `
[
  {
    "inputs": [
      {
        "css": "<unknown>",
        "hasBOM": false,
        "id": "<input css [ID]>",
      },
    ],
    "isColor": false,
    "isHex": false,
    "isUrl": false,
    "isVariable": true,
    "raws": {},
    "source": {
      "end": {
        "column": 9,
        "line": 1,
        "offset": 8,
      },
      "inputId": 0,
      "start": {
        "column": 1,
        "line": 1,
        "offset": 0,
      },
    },
    "type": "word",
    "value": "--batman",
  },
]
`;

exports[`variable parsing > should parse: --main-bg-color 1`] = `"--main-bg-color"`;

exports[`variable parsing > should parse: --main-bg-color 2`] = `"--main-bg-color"`;

exports[`variable parsing > should parse: --main-bg-color 3`] = `
[
  {
    "inputs": [
      {
        "css": "<unknown>",
        "hasBOM": false,
        "id": "<input css [ID]>",
      },
    ],
    "isColor": false,
    "isHex": false,
    "isUrl": false,
    "isVariable": true,
    "raws": {},
    "source": {
      "end": {
        "column": 16,
        "line": 1,
        "offset": 15,
      },
      "inputId": 0,
      "start": {
        "column": 1,
        "line": 1,
        "offset": 0,
      },
    },
    "type": "word",
    "value": "--main-bg-color",
  },
]
`;
